Tests/SonarScanner.MSBuild.PreProcessor.Test/ArchiveDownloaderTests.cs

```diff
@@ -132,24 +132,24 @@ public async Task Download_DownloadFileExists_ChecksumInvalid_DeletesAndReDownlo
     {
         runtime.Directory.Exists(SonarCache).Returns(true);
         runtime.Directory.Exists(ShaPath).Returns(true);
-        runtime.File.Exists(Path.Combine(ShaPath, "javaPath")).Returns(true);
+        runtime.File.Exists(Path.Combine(ShaPath, "target file")).Returns(true);
         runtime.File.Exists(Path.Combine(ShaPath, "filename.tar.gz")).Returns(true);
 
         runtime.File.Create(null).ReturnsForAnyArgs(new MemoryStream(new byte[3], writable: true));
         checksum.ComputeHash(null).ReturnsForAnyArgs(x => "notValid", x => "sha256");
```

Comment:
Use the expected parameter values for the setup. It makes the UT clearer and is also an indirect `Received(1)` test. The lambdas are not needed (https://nsubstitute.github.io/help/multiple-returns/).
```cs
var expectedTarget = Path.Combine(ShaPath, "target file");
runtime.File.Create(expectedTarget).Retruns(...);
checksum.ComputeHash(expectedTarget).Retruns("notValid", "sha256");
```